### PR TITLE
feat(VDataTable): add class to header

### LINKED ISSUE #16648: [Feature Request] V-data-Table-Server ability to add class to header
### Problem to solve
It extend current structure for table headers
Right now we have the next example
https://i.imgur.com/sXF1ZHf.png

In my case I would like to set "class" for specific tr and set color, background or make 
"white-space: nowrap;" or other example. How you will consider as an improvement.
```
const headers = [
  { title: '#', key: 'id' },
  { title: t('messages.code'), key: 'code' },
  { title: t('messages.discount'), key: 'discount' },
  { title: t('messages.use'), key: 'use', width: 'auto' },
  { title: t('messages.short_description'), key: 'short_description', sortable: false },
  { title: t('messages.rules'), key: 'rules', sortable: false },
  { title: t('messages.created_at'), key: 'created_at', **class: 'white-space: nowrap;'** },
  { title: t('messages.expire_at'), key: 'expire_at' }
]
```

### Proposed solution
Current approach to reach the same is use css and find specific row and apply css rule. But my offer make it more attractive and logical 1 place 1 logic

### PR DESCRIPTION
**Pull request by Yumamama00**
## Description
Closes #16648 

## Markup:

```vue
<template>
  <v-app>
    <v-container>
      <v-data-table
        :headers="headers"
        :items="desserts"
        item-value="name"
      />
    </v-container>
  </v-app>
</template>

<script setup>
  const headers = [
    {
      title: 'Dessert (100g serving)',
      align: 'start',
      key: 'name',
      sortable: false,
      class: 'text-primary',
    },
    { title: 'Calories', key: 'calories', class: ['bg-grey-lighten-3', 'text-secondary'] },
    { title: 'Fat (g)', key: 'fat' },
  ]

  const desserts = [
    {
      name: 'Frozen Yogurt',
      calories: 159,
      fat: 6.0,
    },
    {
      name: 'Ice cream sandwich',
      calories: 237,
      fat: 9.0,
    },
    {
      name: 'Eclair',
      calories: 262,
      fat: 16.0,
    },
  ]
</script>

```

### CODE REVIEW

**Comment by scscgit** [comment]
Also what about `cellClass`?
* #17515

File: packages/vuetify/src/labs/VDataTable/VDataTableHeaders.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { useSelection } from './composables/select'
 import { useSort } from './composables/sort'
 
 // Utilities
-import { convertToUnit, genericComponent, useRender } from '@/util'
+import { convertToUnit, genericComponent, useRender, wrapInArray } from '@/util'
```
Comment (johnleider) [inline packages/vuetify/src/labs/VDataTable/VDataTableHeaders.tsx:15]:
```suggestion
import { convertToUnit, genericComponent, useRender } from '@/util'
```

File: packages/vuetify/src/labs/VDataTable/VDataTableHeaders.tsx [hunk 2]
```diff
@@ -85,6 +85,7 @@ export const VDataTableHeaders = genericComponent<VDataTableHeadersSlots>()({
               'v-data-table__th--sorted': isSorted,
             },
             loaderClasses.value,
+            ...wrapInArray(column.class),
```
Comment (johnleider) [inline packages/vuetify/src/labs/VDataTable/VDataTableHeaders.tsx:88]:
```suggestion
            column.class,
```

File: packages/vuetify/src/labs/VDataTable/types.ts [hunk 3]
```diff
@@ -13,6 +13,7 @@ export type DataTableHeader = {
 
   fixed?: boolean
   align?: 'start' | 'end'
+  class?: string | string[]
```
Comment (johnleider) [inline packages/vuetify/src/labs/VDataTable/types.ts:16]:
```suggestion
  class?: any
```

**Review by johnleider — CHANGES_REQUESTED** [review]

**Comment by johnleider** [comment]
Features go to the dev branch

**Comment by Yumamama00** [comment]
Thanks for the review. Fixed.

**Comment by Yumamama00** [comment]
> Also what about `cellClass`?
> 
> * [[Feature Request] Add custom css class to table headers and table columns  #17515](https://github.com/vuetifyjs/vuetify/issues/17515)

cellClass was addressed separately in the following PR.
https://github.com/vuetifyjs/vuetify/pull/18038

**Comment by websitevirtuoso** [comment]
Long time without feedback. Can we check it? @johnleider

**Comment by johnleider** [comment]
Anything not tagged for v3.4 milestone is very low priority atm.

**Comment by KaelWD** [comment]
Replaced by #18589

**Pull request closed without merging**